src/utils.py

```diff
@@ -297,51 +331,47 @@ def run_policy_until(policy: Callable[[State], Action],
 
 def run_policy_on_task(
     policy: Callable[[State], Action],
-    task: Task,
-    simulator: Callable[[State, Action], State],
+    env: BaseEnv,
+    train_or_test: str,
+    task_idx: int,
     max_num_steps: int,
-    render: Optional[Callable[[State, Task, Optional[Action]],
-                              List[Image]]] = None,
+    make_video: bool = False,
 ) -> Tuple[LowLevelTrajectory, Video, bool]:
     """A light wrapper around run_policy_until that takes in a task and uses
     achieving the task's goal as the termination_function.
 
     Returns the trajectory and whether it achieves the task goal. Also
     optionally returns a video, if a render function is provided.
     """
+    task = env.get_task(train_or_test, task_idx)
 
     def _goal_check(state: State) -> bool:
         return all(goal_atom.holds(state) for goal_atom in task.goal)
 
-    traj = run_policy_until(policy, simulator, task.init, _goal_check,
+    traj = run_policy_until(policy, env, train_or_test, task_idx, _goal_check,
                             max_num_steps)
     goal_reached = _goal_check(traj.states[-1])
     video: Video = []
-    if render is not None:  # step through the traj again, making the video
+    if make_video:  # step through the traj again, making the video
         for i, state in enumerate(traj.states):
             act = traj.actions[i] if i < len(traj.states) - 1 else None
-            video.extend(render(state, task, act))
+            video.extend(env.render(state, task, act))
     return traj, video, goal_reached
 
 
-def policy_solves_task(policy: Callable[[State], Action], task: Task,
-                       simulator: Callable[[State, Action], State]) -> bool:
-    """A light wrapper around run_policy_on_task that returns whether the given
-    policy solves the given task."""
-    _, _, solved = run_policy_on_task(policy, task, simulator,
-                                      CFG.max_num_steps_check_policy)
-    return solved
-
-
 def option_to_trajectory(init_state: State,
                          simulator: Callable[[State, Action],
                                              State], option: _Option,
                          max_num_steps: int) -> LowLevelTrajectory:
     """A light wrapper around run_policy_until that takes in an option and uses
-    achieving its terminal() condition as the termination_function."""
+    achieving its terminal() condition as the termination_function.
```

Comment:
Similar to `run_policy_on_task` -- we should really just get rid of this method and substitute in these two lines. That way it's easy to remember: `run_policy_until` = good, `simulate_policy_until` = bad.